test/trento/activity_log_test.exs

```diff
@@ -0,0 +1,180 @@
+defmodule Trento.ActivityLogTest do
+  @moduledoc false
+  use ExUnit.Case
+  use Trento.DataCase
+
+  import Mox
+
+  import Trento.Factory
+
+  require Trento.ActivityLog.RetentionPeriodUnit, as: RetentionPeriodUnit
+
+  alias Trento.ActivityLog
+  alias Trento.ActivityLog.RetentionTime
+  alias Trento.ActivityLog.Settings
+
+  setup :verify_on_exit!
+
+  describe "retrieving activity log settings" do
+    test "should return an error when settings are not available" do
+      assert {:error, :activity_log_settings_not_configured} == ActivityLog.get_settings()
+    end
+
+    test "should return settings" do
+      %{
+        retention_time: %{
+          retention_period: retention_period,
+          retention_period_unit: retention_period_unit
+        }
+      } = insert(:activity_log_settings)
+
+      assert {:ok,
+              %Settings{
+                retention_time: %RetentionTime{
+                  retention_period: ^retention_period,
+                  retention_period_unit: ^retention_period_unit
+                }
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->

1. **thought:** ‏I just realize that this result may be a little verbose. Should we prefer a more lean structure such as
```
{
  retention_time: {
    value: 10
    unit: "month"
  }
}
```

I don't have a strong opinion nor I am considering the needed renaming in the model, I'm just thinking on that

Reply:
Sure 👍 gonna change shortly. It's actually nicer, thanks!

Reply:
here we go 9432302d0ff1036ccb247a81b19be8beae95fe1f